src/e2e/java/teammates/e2e/pageobjects/BrowserPool.java

```diff
@@ -39,19 +43,27 @@ private static BrowserPool getInstance() {
     /**
      * Returns a Browser object ready to be used.
      */
-    public static Browser getBrowser() {
+    public static Browser getBrowser(String name) {
+        if (TestProperties.BROWSER_SAUCELABS.equals(TestProperties.BROWSER)) {
+            return new Browser(name);
+        }
         return getInstance().requestInstance();
     }
 
     /**
      * Releases a Browser instance back to the pool, ready to be reused.
      */
-    public static void release(Browser browser) {
-        BrowserPool pool = getInstance();
-        // Synchronized to ensure thread-safety
-        synchronized (pool) {
-            browser.isInUse = false;
-            pool.notifyAll();
+    public static void release(Browser browser, boolean isSuccess) {
+        if (TestProperties.BROWSER_SAUCELABS.equals(TestProperties.BROWSER)) {
+            ((JavascriptExecutor) browser.driver).executeScript("sauce:job-result=" + (isSuccess ? "passed" : "failed"));
```

Comment:
I guess here we want to notify sauce lab about the result of the execution of the test?

Maybe we want to put this in https://github.com/TEAMMATES/teammates/pull/9597/files#diff-5ac8e80833810cd52d80b7f75531503dR62 instead. After all, browser should have no idea about whether a test is success or not (so it is a bit confusing to have a `isSuccess` parameter in the method signature)?